prov/util/src/util_av.c

```diff
@@ -1670,6 +1670,12 @@ void ofi_cmap_free(struct util_cmap *cmap)
 		util_cmap_del_handle(peer->handle);
 	}
 	util_cmap_cm_thread_close(cmap);
+	/* cleanup function would be used in manual progress mode */
+	if (cmap->attr.cleanup) {
+		cmap->release(&cmap->lock);
+		cmap->attr.cleanup(cmap->ep);
+		cmap->acquire(&cmap->lock);
```

Comment:
Releasing and re-acquiring the lock is racy.  Please describe why this wouldn't be the case.

Reply:
The cleanup function would acquire cmap->lock when processing any pending events in the manual progress event list. So we need to release the lock here.

Also, we could just release it and not acquire it again as the CM thread won't be running anymore.